pyston/resource.py

```diff
@@ -641,6 +639,28 @@ def get_order_fields_rfs(self):
         else:
             return rfs(extra_order_fields).join(rfs(order_fields))
 
+    def get_methods_returning_field_value(self, fields):
+        """
+        Returns dict of resource methods which can be used with serializer to get a field value.
+        :param fields: list of field names
+        :return: dict of resource methods. Key is a field name, value is a method that returns field value.
+        """
+        method_fields = {}
+        for field_name in fields:
+            method = self.get_method_returning_field_value(field_name)
+            if method:
+                method_fields[field_name] = method
+        return method_fields
+
+    def get_method_returning_field_value(self, field_name):
+        """
+        Returns method which can be used with serializer to get a field value.
+        :param field_name: name of th field
+        :return: resource method
+        """
+        method = getattr(self, field_name, None)
+        return method if method and callable(method) else None
```

Comment:
vrátí metodu resource, pokud je callcable. Přemýšlím nad nějakým systémem kontroly, které metody mohou být z resource využity a které ne. Ještě to nemám ale úplně dopiplané, takže zatím vlastně jen refaktoruji.